packages/compiler/core/messages.ts

```diff
@@ -701,8 +701,8 @@ const diagnostics = {
     severity: "error",
     messages: {
       default: "Invalid encoding",
-      wrongType: paramMessage`Encoding '${"encoding"}' cannot be used on type ${"type"}. Expected a ${"expected"}.`,
-      wrongEncodingType: paramMessage`Encoding '${"encoding"}' cannot be used on type ${"type"}. Expected a ${"expected"}.`,
+      wrongType: paramMessage`Encoding '${"encoding"}' cannot be used on type '${"type"}'. Expected: ${"expected"}.`,
+      wrongEncodingType: paramMessage`Encoding '${"encoding"}' cannot be used on type '${"type"}'. Expected '${"expected"}', but got '${"actual"}'.`,
```

Comment:
I think this message is overall still confusing, think the point we trying to say here is "that the type that it will be serialized as is expected to be XYZ but it was ABC for this encoding"

Reply:
`Encoding 'rfc3339' on type 's' is expected to be serialized as 'string' but got 'int32'.`
Here's the example I updated it to. Is this clearer?